### PR TITLE
bookTreeView tests

### PR DESCRIPTION
**Pull request by MaddyDev**
Tests for missing methods in BooksTreeView.

### CODE REVIEW

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -231,19 +229,17 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
...
 
 				if (this._visitedNotebooks.indexOf(normalizedResource) === -1
 					&& this._bookTrustManager.isNotebookTrustedByDefault(normalizedResource)) {
-					let openDocumentListenerUnsubscriber = azdata.nb.onDidOpenNotebookDocument((document: azdata.nb.NotebookDocument) => {
-						document.setTrusted(true);
-						this._visitedNotebooks = this._visitedNotebooks.concat([normalizedResource]);
-						openDocumentListenerUnsubscriber.dispose();
-					});
+					let document = azdata.nb.notebookDocuments.find(document => document.fileName === resource);
+					document.setTrusted(true);
```
Comment (corivera) [inline extensions/notebook/src/book/bookTreeView.ts:240]: Do we need to check for undefined here?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 2]
```diff
@@ -455,35 +447,9 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		return { notebookPaths: notebookPaths, bookPaths: bookPaths };
 	}
 
-	private runThrottledAction(resource: string, action: () => void) {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:458]: Do we know why this was thought to be necessary to start with?

File: extensions/notebook/src/test/book/book.test.ts [hunk 3]
```diff
@@ -567,6 +569,35 @@ describe('BooksTreeViewTests', function () {
...
+		});
+
+		it('openNotebook should open notebook in the editor', async () => {
+			let notebookPath = path.join(rootFolderPath, 'content', 'notebook2.ipynb');
+			await bookTreeViewProvider.openNotebook(notebookPath);
+			should(azdata.nb.notebookDocuments.length).equal(1, 'Should have opened the notebook in the editor.');
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		});
+
+		it('openNotebookAsUntitled should open a notebook as untitled file in the editor', async () => {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:587]: These test names don't seem to match up - for example this is openNotebookAsUntitled but it appears to be testing openExternalLink?

File: extensions/notebook/src/book/bookTreeView.ts (hunk 2 above)
Comment (MaddyDev) [reply extensions/notebook/src/book/bookTreeView.ts:458]: No, it was there from when I took over. @chlafreniere  do you have any context here?

File: extensions/notebook/src/test/book/book.test.ts (hunk 3 above)
Comment (MaddyDev) [reply extensions/notebook/src/test/book/book.test.ts:587]: Thanks Charles for catching this, there was some mix up with copy paste and the tests for jumbled.

File: extensions/notebook/src/book/bookTreeView.ts (hunk 2 above)
Comment (chlafreniere) [reply extensions/notebook/src/book/bookTreeView.ts:458]: https://github.com/microsoft/azuredatastudio/pull/6420

Comment (chlafreniere) [reply extensions/notebook/src/book/bookTreeView.ts:458]: There's some description in the PR. Please check if that behavior still exists, where multiple calls are coming in.

File: extensions/notebook/src/test/book/book.test.ts [hunk 4]
```diff
@@ -567,6 +571,35 @@ describe('BooksTreeViewTests', function () {
 			should(notebookUri.scheme).equal('untitled', 'Failed to get untitled uri of the resource');
 		});
 
+		it('openMarkdown should open markdown in the editor', async () => {
+			let executeCommandSpy = sinon.spy(vscode.commands, 'executeCommand');
+			let notebookPath = path.join(rootFolderPath, 'content', 'readme.md');
+			bookTreeViewProvider.openMarkdown(notebookPath);
+			should(executeCommandSpy.calledWith('markdown.showPreview')).be.true('openMarkdown should have called markdown.showPreview');
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:579]: Cleanup like this should either be in a finally, or as a afterEach so that we don't mess up other tests if these tests fail and leave stale state.

Comment (Charles-Gagnon) [reply extensions/notebook/src/test/book/book.test.ts:579]: You're currently calling `await vscode.commands.executeCommand('workbench.action.closeAllEditors');` in the afterAll - moving that to afterEach seems like a safer way of handling all this.

File: extensions/notebook/src/test/book/book.test.ts [hunk 5]
```diff
@@ -533,6 +538,34 @@ describe('BooksTreeViewTests', function () {
...
 
+		it('openMarkdown should open markdown in the editor', async () => {
+			let executeCommandSpy = sinon.spy(vscode.commands, 'executeCommand');
+			let notebookPath = path.join(rootFolderPath, 'content', 'readme.md');
+			bookTreeViewProvider.openMarkdown(notebookPath);
+			should(executeCommandSpy.calledWith('markdown.showPreview')).be.true('openMarkdown should have called markdown.showPreview');
+		});
+
+		// TODO: Need to investigate why it's failing on linux.
+		it.skip('openNotebook should open notebook in the editor', async () => {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:549]: Why did you change this? Having unstable tests run in the unstable runs is potentially still useful so we should use that over skip when we can.

Comment (chlafreniere) [reply extensions/notebook/src/test/book/book.test.ts:549]: Unstable tests should be unstable. I'm not convinced that tests that always fail should be run as part of "unstable"

Comment (Charles-Gagnon) [reply extensions/notebook/src/test/book/book.test.ts:549]: It fails 100% of the time? I mean to me if that's the case that sounds like a concern we should investigate now then...the test is testing a pretty basic scenario and if that isn't working we have some big issues

Comment (chlafreniere) [reply extensions/notebook/src/test/book/book.test.ts:549]: Lemme try to help out here to unblock. Because it actually fails for me when i un-skip the test on my Mac.
